command/issues/list/list.go

```diff
@@ -145,11 +131,34 @@ func (opts *IssuesListOptions) getIssuesData(ctx context.Context) (err error) {
 
 	// Fetch issues for a certain FileArg (filepath) passed by the user
 	// Example: `deepsource issues list api/hello.py`
-	if opts.FileArg != "" {
-		opts.issuesData, err = deepsource.GetIssuesForFile(ctx, opts.SelectedRemote.Owner, opts.SelectedRemote.RepoName, opts.SelectedRemote.VCSProvider, opts.FileArg, opts.LimitArg)
+	if len(opts.FileArg) != 0 {
+		var fetchedIssues []issues.Issue
+		var filteredIssues []issues.Issue
+
+		opts.issuesData, err = deepsource.GetIssues(ctx, opts.SelectedRemote.Owner, opts.SelectedRemote.RepoName, opts.SelectedRemote.VCSProvider, opts.LimitArg)
 		if err != nil {
 			return err
 		}
+
+		for _, arg := range opts.FileArg {
+			// use regex to segregate directory and file names
+			var pathRegexp = regexp.MustCompile(`^(.*/)?(?:$|(.+?)(?:(\.[^.]*$)|$))`)
+			matched := pathRegexp.FindStringSubmatch(arg)
+
+			// if the argument contains a directory, fetch issues for the directory
+			if matched[1] != "" {
+				filteredIssues, err = filterIssuesByPath(arg, opts.issuesData)
+				if err != nil {
+					return err
+				}
+
+				fetchedIssues = append(fetchedIssues, filteredIssues...)
+				continue
+			}
+		}
```

Comment:
```suggestion
			// If the argument doesn't contain a directory or file, skip it
			if matched[1] == "" {
			  continue
			}
			// Filter issues for the valid directories/files
			filteredIssues, err = filterIssuesByPath(arg, opts.issuesData)
			if err != nil {
				return err
			}
			fetchedIssues = append(fetchedIssues, filteredIssues...)
	  }
```

Reply:
Remember, guard clauses. Try to stay away from big `if-else` blocks.

Reply:
Fixed.